_posts/2019-06-06-elixir-conf-eu-2019-review.md

```diff
@@ -0,0 +1,192 @@
+
+As a bonus, we get to see how they have been already playing with Broadway, and how easy to swap it in instead of Flow, along the differences that they have as systems.
+
+[![flow](https://img.youtube.com/vi/t46L9RKmlNo/0.jpg)](https://www.youtube.com/watch?v=t46L9RKmlNo)
+
+### Telemetry ...and metrics for all - ElixirConf EU 2019 by Arkadiusz Gil
+
+Arkadiusz Gil's talk was about Telemetry, a dynamic dispatching library used for metrics and instrumentations.
+
+He started by talking about monitoring, and how crucial it is to improving performance. The goal is to use monitoring early on in the lifecycle of our project (as we have done already with testing and deployment), so we can use it as another form of verification. Next, he showed us how we can do metrics today, using custom functions (like the number of requests, the number of successful responses, how big a load we are pushing to external systems like the the database, how to track memory or cpu usage from the vm etc). 
+
+But as we add more and more, this custom set up becomes repetitive. Telemetry uses emitted events from various parts of our systems, and then attaches to those events by using handlers, which are in turn pushed to a reporter system like statsd. 
+
+A Telemetry event consists for three parts, the event name, the measurements (measurable properties like payload) and some metadata. Various libraries are beginning to add support for Telemetry emitting events in them, like Ecto from v3.1, Phoenix from v1.5 and Plug from 1.8. Arkadiusz also showed us `Telemetry.Poller` which every couple of seconds picks some metrics from the BEAM and emits them as Telemetry events and `Telemetry.Metrics` which allows us to specify how telemetry events are aggregated over time. Some examples of metrics is `last_value`, `sum`, `counter` and `distribution` which gives us some insight into statistics. 
+
+Lastly, he demonstrated the way you can plug Telemetry into a project by using the Rumble application from the Phoenix Programming book. It is as simple as adding the dependencies in `mix.exs`, making a custom module that defines which metrics we want, supervising it in `application.ex` and we are good to go from the application standpoint. The difficult part is setting up the reporter, like statsd, but there is going to be one included in Phoenix as well, where we will be able to access it from a route.
+
+[![](https://img.youtube.com/vi/cOuyOmcDV1U/0.jpg)](https://www.youtube.com/watch?v=cOuyOmcDV1U)
+
+## Erlang Ecosystem Foundation
+
+As they did in CODE BEAM SF conference, we were given the presentation on the new [Erlang Ecosystem Foundation](https://erlef.org/). They stated again that things are still a little blurry, and we will probably have more information in the coming months, but check out the video of the presentation just to know what's coming.
```

Comment:
should you add the yt video of their presentation here?

Reply:
It hasn't been uploaded yet. In general, they have taken their time uploading the videos for that conference, Idk why.